reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -586,6 +590,22 @@ public final SPEC maxLifeTime(Duration maxLifeTime) {
 			return get();
 		}
 
+		/**
+		 * Set the options to use for configuring {@link ConnectionProvider} eviction predicate.
+		 * Default to null implying connections will not be evaluated for eviction. The eviction predicate works
+		 * alongside {@link #maxLifeTime(Duration)} and {@link #maxIdleTime(Duration)}.
+		 * <p><strong>Note:</strong> This configuration is not applicable for {@link reactor.netty.tcp.TcpClient}.
+		 * A TCP connection is always closed and never returned to the pool.
+		 *
+		 * @param evictionPredicate The predicate function that evaluates whether a connection should be evicted
+		 * @return {@literal this}
+		 * @throws NullPointerException if evictionPredicate is null
+		 */
+		public final SPEC evictionPredicate(BiPredicate<Connection, PooledRefMetadata> evictionPredicate) {
```

Comment:
@samueldlightfoot We don't expose Reactor Pool API directly so we need something provided by Reactor Netty. Check `AllocationStrategy` ...